VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/waiting-room-shared/waiting-room-base.component.ts

```diff
@@ -923,15 +923,22 @@ export abstract class WaitingRoomBaseDirective {
         this.videoCallService.stopScreenShare();
     }
 
-    get shouldMuteHearing(): boolean {
-        return !(
-            (this.countdownComplete &&
-                this.participant.status === ParticipantStatus.InHearing &&
-                this.hearing.status === ConferenceStatus.InSession) ||
-            (this.participant.status === ParticipantStatus.InConsultation && !this.hasTriedToLeaveConsultation)
+    shouldMuteHearing(): boolean {
+        return !(this.shouldUnmuteForHearing() || this.shouldUnmuteForConsultation());
+    }
+
+    shouldUnmuteForHearing(): boolean {
+        return (
+            this.countdownComplete &&
+            this.participant.status === ParticipantStatus.InHearing &&
+            this.hearing.status === ConferenceStatus.InSession
         );
     }
 
+    shouldUnmuteForConsultation(): boolean {
+        return this.participant.status === ParticipantStatus.InConsultation && !this.hasTriedToLeaveConsultation;
+    }
```

Comment:
Splitting the existing logic into 2 separate methods to make it clearer what the cases are and so that I can extend the functionality individually.